### PR TITLE
Modal: Apply any preexisting body padding again after closing

### PR DESCRIPTION
**Pull request by hnrch02**
Modifies #15273 to ignore padding set via CSS.

### CODE REVIEW

File: js/modal.js [hunk 1]
```diff
@@ -259,11 +260,12 @@
 
   Modal.prototype.setScrollbar = function () {
     var bodyPad = parseInt((this.$body.css('padding-right') || 0), 10)
+    this.originalBodyPad = this.$body.attr('style') ? bodyPad : null
```
Comment (cvrebert) [inline js/modal.js]: Won't this break with something like:
```
body { padding-right: 5px; }
```
```html
<body style="color: red;">
````
?

Comment (hnrch02) [inline js/modal.js]: Yup, that's why I haven't merged it yet. Was thinking about checking it with `/padding-right/i`, what do you think?

Comment (cvrebert) [inline js/modal.js]: What about `!!this.$body[0].style.paddingRight` ?

Comment (hnrch02) [inline js/modal.js]: Much better - I keep forgetting that `Element.style` does not include computed styles.

**Comment by hnrch02** [comment]
@cvrebert LGTY now? Added another unit test for the case you described.

**Comment by cvrebert** [comment]
What about `<body style="padding-right: 5%">`? I think the current code would replace `5%` with its calculated pixel value when doing the restoration.

**Comment by hnrch02** [comment]
Good catch. If only I was as wise as you are :bowtie:

**Comment by cvrebert** [comment]
What would programming be without edge cases? :-P

File: js/modal.js [hunk 2]
```diff
@@ -259,11 +260,12 @@
 
   Modal.prototype.setScrollbar = function () {
     var bodyPad = parseInt((this.$body.css('padding-right') || 0), 10)
+    this.originalBodyPad = document.body.style.paddingRight || null
```
Comment (cvrebert) [inline js/modal.js]: What about `|| ''` instead of `|| null`, and then getting rid of the ternary below?

**Comment by hnrch02** [comment]
Nothing of interest, certainly :smile:

File: js/modal.js (hunk 2 above)
Comment (hnrch02) [inline js/modal.js]: I'm getting a little tired, that I could have noticed.

File: js/tests/unit/modal.js [hunk 3]
```diff
@@ -301,4 +301,81 @@ $(function () {
...
+        $style.remove()
+        done()
+      })
+      .on('shown.bs.modal', function () {
+        $(this).bootstrapModal('hide')
+      })
+      .bootstrapModal('show')
+  })
+
+  QUnit.test('should ignore values set via CSS when trying to restore body padding after closing', function (assert) {
```
Comment (cvrebert) [inline js/tests/unit/modal.js]: This test description needs to be updated

**Comment by cvrebert** [comment]
Once that description is fixed, :+1: :100: :shipit:

**Pull request merged**